### PR TITLE
Rename `iot_dps` resources to `iothub_dps` and add missing properties

### LINKED ISSUE #4879: Add missing properties to azurerm_iot_dps
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

The following is provided by the Azure REST API and should be returned by the **azurerm_iot_dps** resource:

```
{
  "etag": "....",
  "name": "example_dps_service",
  "location": "A Fun location",
  "properties": {
    "state": "Active",
    "iotHubs": [
      {
        .....
      }
    ],
    "allocationPolicy": "static",
    "serviceOperationsHostName": "example_dps_service.azure-devices-provisioning.net",
    "deviceProvisioningHostName": "deviceglobal.azure-devices-provisioning.net",
    "idScope": "a_sample_id_scope"
  },
```

Add the **azurerm_iot_dps** missing properties:

- service_operations_host_name
- device_provisioning_host_name
- id_scope
- allocation_policy

### New or Affected Resource(s)

- azurerm_iot_dps

### Potential Terraform Configuration

Add the following properties to the outputs:
- service_operations_host_name
- device_provisioning_host_name
- id_scope
- allocation_policy

```hcl
			"service_operations_host_name": {
				Type:     schema.TypeString,
				Computed: true,
			},

			"device_provisioning_host_name": {
				Type:     schema.TypeString,
				Computed: true,
			},

			"id_scope": {
				Type:     schema.TypeString,
				Computed: true,
			},

			"allocation_policy": {
				Type:     schema.TypeString,
				Computed: true,
			},
```

### References

* [GET Example in Azure REST API](https://docs.microsoft.com/en-us/rest/api/iot-dps/iotdpsresource/get#examples)

### PR DESCRIPTION
**Pull request by brunhil**
- Rename Iot Hub resources to be consistent
    - azurerm_iot_dps --> azurerm_iothub_dps
    - azurerm_iot_dps_certificate --> azurerm_iothub_dps_certificate
- Consolidates all IoT Hub pages under an IoT Hub group in the website for providers
- Fixes #4879
    - Adds following properties as outputs:
        - service_operation_host_name
        - device_provisioning_host_name
        - id_scope
        - allocation_policy

Abandoned PR 4881 as I had rebased from master and messed up history. This pulls in only the changes from that branch :).

### CODE REVIEW

File: azurerm/provider.go [hunk 1]
```diff
@@ -297,8 +297,8 @@ func Provider() terraform.ResourceProvider {
 		"azurerm_hdinsight_storm_cluster":                            resourceArmHDInsightStormCluster(),
 		"azurerm_healthcare_service":                                 resourceArmHealthcareService(),
 		"azurerm_image":                                              resourceArmImage(),
-		"azurerm_iot_dps":                                            resourceArmIotDPS(),
-		"azurerm_iot_dps_certificate":                                resourceArmIotDPSCertificate(),
```
Comment (katbyte) [inline azurerm/provider.go]: So we typically leave in the old resources with a deprecation notice so there is no breaking changes

File: azurerm/resource_arm_iothub_dps.go [hunk 2]
```diff
@@ -139,12 +139,32 @@ func resourceArmIotDPS() *schema.Resource {
...
+
+			"id_scope": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"allocation_policy": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
```
Comment (katbyte) [inline azurerm/resource_arm_iothub_dps.go]: would we arrange these in alphabetical order?

File: website/docs/r/iothub_dps.html.markdown [hunk 3]
```diff
@@ -78,10 +78,18 @@ The following attributes are exported:
 
 * `id` - The ID of the IoT Device Provisioning Service.
 
+* `service_operations_host_name` - The service endpoint of the IoT Device Provisioning Service.
+
+* `device_provisioning_host_name` - The device endpoint of the IoT Device Provisioning Service.
+
+* `id_scope` - The unique identifier of the IoT Device Provisioning Service.
+
+* `allocation_policy` - The allocation policy of the IoT Device Provisioning Service.
```
Comment (katbyte) [inline website/docs/r/iothub_dps.html.markdown]: could we order these alphabetically after the id property?

File: azurerm/resource_arm_iot_dps_certificate_test.go [hunk 4]
```diff
@@ -183,10 +183,10 @@ resource "azurerm_iot_dps" "test" {
   }
 }
 
-resource "azurerm_iot_dps_certificate" "test" {
+resource "azurerm_iothub_dps_certificate" "test" {
```
Comment (katbyte) [inline azurerm/resource_arm_iot_dps_certificate_test.go]: another one, and more in the rest of the file

File: website/docs/guides/migrating-between-renamed-resources.html.markdown [hunk 5]
```diff
@@ -9,20 +9,17 @@ description: |-
```
Comment (brunhil) [inline website/docs/guides/migrating-between-renamed-resources.html.markdown]: I went a head and made changes here to support future versions. Let me know your thoughts @katbyte

**Pull request merged**